operator/pkg/helmreconciler/wait.go

```diff
@@ -0,0 +1,255 @@
+package helmreconciler
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+	"k8s.io/api/apps/v1"
+	v12 "k8s.io/api/core/v1"
+	"k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	apiextensionsclient "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	v13 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	deployment2 "k8s.io/kubectl/pkg/util/deployment"
+
+	"istio.io/istio/operator/pkg/name"
+	"istio.io/istio/operator/pkg/object"
+	"istio.io/istio/operator/pkg/util/clog"
+)
+
+const (
+	// cRDPollInterval is how often the state of CRDs is polled when waiting for their creation.
+	cRDPollInterval = 500 * time.Millisecond
+	// cRDPollTimeout is the maximum wait time for all CRDs to be created.
+	cRDPollTimeout = 60 * time.Second
+)
+
+// deployment holds associated replicaSets for a deployment
+type deployment struct {
+	replicaSets *v1.ReplicaSet
+	deployment  *v1.Deployment
+}
+
+// WaitForResources polls to get the current status of all pods, PVCs, and Services
+// until all are ready or a timeout is reached
+func WaitForResources(objects object.K8sObjects, restConfig *rest.Config, cs kubernetes.Interface, waitTimeout time.Duration, dryRun bool, l clog.Logger) error {
+	if dryRun {
+		l.LogAndPrint("Not waiting for resources ready in dry run mode.")
+		return nil
+	}
+
+	if err := waitForCRDs(objects, restConfig); err != nil {
```

Comment:
nit: can this use `cs` instead of `restConfig`?

Reply:
i don't think so because waitForCRDs needs an apiextensions client.